resource.go

```diff
@@ -4,40 +4,30 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/qor/qor"
 	"github.com/qor/qor/admin"
 )
 
-type Resource struct {
-	*admin.Resource
-	Record  interface{}
-	Context *qor.Context
-}
-
-func New(res *admin.Resource, record interface{}, context *qor.Context) *Resource {
-	return &Resource{Resource: res, Record: record, Context: context}
-}
-
-func (res Resource) getPrimaryKey() string {
-	db := res.Context.GetDB()
+func getPrimaryKey(context *admin.Context) string {
+	db := context.GetDB()
 
 	var primaryValues []string
-	for _, field := range db.NewScope(res.Record).PrimaryFields() {
+	result, _ := context.FindOne()
+	for _, field := range db.NewScope(result).PrimaryFields() {
 		primaryValues = append(primaryValues, fmt.Sprint(field.Field.Interface()))
 	}
 	return strings.Join(primaryValues, "::")
 }
 
-func (res Resource) CreateActivity(activity QorActivity) error {
-	db := res.Context.GetDB()
-	activity.ResourceType = res.ToParam()
-	activity.ResourceID = res.getPrimaryKey()
+func CreateActivity(activity QorActivity, context *admin.Context) error {
+	db := context.GetDB()
+	activity.ResourceType = context.Resource.ToParam()
+	activity.ResourceID = getPrimaryKey(context)
 	return db.Save(&activity).Error
 }
 
-func (res Resource) GetActivities(types ...string) ([]QorActivity, error) {
+func GetActivities(context *admin.Context, types ...string) ([]QorActivity, error) {
```

Comment:
It is better to pass current result into get activities, so don't need to find it again from the `getPrimaryKey` func?